src/App.tsx

```diff
@@ -1,8 +1,78 @@
-import React from 'react';
+import React, { useState} from 'react';
+import { getUsers, getPosts, Post, getComments } from './helper/api';
+import debounce from 'lodash/debounce';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+const postsFromFiler = (posts: Post[], qery: string) =>{
+  return posts.filter( post => post.title.includes(qery) || post.body.includes(qery))
+}
+
+const App = () => {
+  const [posts, setPost] = useState<Post[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [qery, setQery] = useState('');
+  const [filterQuery, setFilterQuery] = useState('');
+const setFilterQueryDebounce = debounce(setFilterQuery, 1000)
```

Comment:
indentation

Reply:
check the page, it doesn't work
try to add and remove one letter quickly and see how slow the content gets